bench/Bench/Data/Map.purs

```diff
@@ -43,6 +77,50 @@ benchMap = do
     log $ "size: big map (" <> show (M.size bigMap) <> ")"
     benchWith 10  \_ -> M.size bigMap
 
+  benchFoldable = do
+    let nats = L.range 0 999999
+        natPairs = (\x -> Tuple x x) <$> nats
+        bigMap = Map2a0bff.fromFoldable $ natPairs
+        bigMap' = M.fromFoldable $ natPairs
+        size = Map2a0bff.size bigMap
+        size' = M.size bigMap'
+
+    log $ "Map2a0bff.foldr big map (" <> show size <> ")"
+    benchWith 10 \_ -> F.foldr (+) 0 bigMap
```

Comment:
If these benchmarks are only to test the folding performance, and not include say dispatch of `<>`, I think these folds should all be `\_ _ -> unit` or `\_ _ _ -> unit`. I know all the work is "consistent" between tests, I'm just not sure it's necessary to include extra work.